### DISCUSSION TITLE
Any ideas about pool design?

### DISCUSSION THREAD

**Discussion by yarnauy** [Chia-Network/chia-blockchain#2169] [Ideas]
Does anyone knows that how existing mining pool works or how to design a mining pool? There are already some pools now, but they do not open source their mining policy yet. That's maybe a bit centralized.  
Or is there some information about pool design, e.g., the architecture、the distribution mechanism...

**Comment by hoffmang9** [comment]
Most all current plots are plotted to self pool. Pools are coming in a few weeks - https://www.chia.net/2020/11/10/pools-in-chia.html

**Reply by tianxinya2021** [reply]
I like farm pool idea. Does that mean we have to re-plot all the plots in order to use farm pool? Thanks.

**Reply by hoffmang9** [reply]
You will only be able to put new plots into a pool.

**Reply by Tuguk** [reply]
Can u explain...it means we should wait an official pool because we will lose our old plots or we can make plots as usual and then add plots for official pool without problem?

**Reply by CharlesWiltgen** [reply]
The latter—you can continue to make plots as usual, then start to add plots for pooling once pools are available.

**Reply by hbroer** [reply]
Hey, I have a question about how pooling will be done. Is there a official pool demon/server with a API for status informations we can use, or do pool provider need to implement against the protocol themself? Is there a documentation on how it has to be done?
